### DISCUSSION TITLE
Can't get a model result like Geomodeller does.

### DISCUSSION THREAD

**Discussion by SpongeGirl** [gempy-project/gempy#888] [General]
I want to do a compare test between gempy and Geomodeller,  I use the lastest release code and follow the "1.1 -Basics of geological modeling with GemPy", the input data i use is the tutorial sample data of Geomodeller that i post below (Drilling.zip).
so this is my question:
1、i think you may be familiar with Geomodeller, i wonder if gempy can get a model like Geomodeller software does, below is the result which i use Geomodeller to create, if so and i hope you can tell me how (cause i've researched a few days about gempy and readed most of the issues and discussions);

![image](https://github.com/gempy-project/gempy/assets/44236813/cb70cfee-a7a8-4735-a1b6-2b8938803fee)

2、i still care about how gempy deal with the data without orientation(it's important), and how G_x/G_y/G_z is compute cause no G_x/G_y/G_z will get error at create_geomodel step, and the method appears in issuses is not in lastest release(like 'create_orientation_from_surface_points').

I'm a new researcher about gempy and i will very appreciate if you can give me some help to solve my problem.
[Drilling.zip](https://github.com/gempy-project/gempy/files/15140576/Drilling.zip)

**Comment by javoha** [comment]
Hello,
can you maybe post specifically the errors/problems you encountered when setting up the model? Looking at the Geomodeller example you posted you definitley need to set up the structural frame correctly.
Regarding your second point: Gempy is based on a universal co-kriging algorithm so it does require orientation data. Maybe take a look at the referenced literature to get a better understanding?
Cheers
Jan

Small edit: I will also for now convert this into a discussion, as I dont think it is neccessarily an issue or bug.

**Comment by SpongeGirl** [comment]
Thanks for your suggestion, I use the orientation data which create in geomodeller and finally it works.
so i wonder if gempy can create (or use the surface points to compute) valid orientation data? Or gempy can only use the orientation data well prepared and just do the visualization job?

**Reply by javoha** [reply]
Great to hear that it worked. If you like it would be nice if you share your model (or at least an image) - would be cool to see it. Regarding the orientations: How does geomodeller infer orientations. Gempy leaves this somewhat open to the user. There might be actual outcrop measurements or similar datasets that are unrelated to the boreholes. 
You can also infer it from borehole data, but you do not need gempy specifically for this. We used to have a method to infer from triangulated surfaces as an approximation, but I guess there is a limitless amount of getting orientations (probably of varying quality).

**Reply by SpongeGirl** [reply]
Haha, actually, the orientations data in geomodeller is added by user through choosing two points from the same layer, not computed by itself, that's why i want to know how gempy gets the orientations, cause it's very important to decide the shape of the final model.
Below is one section from the input data, those small arrows are the orientations, i just copy the values and import to gempy.
![image](https://github.com/gempy-project/gempy/assets/44236813/db3c8a56-f9a0-4cc1-86a7-f2fd901ed23d)
Yes, i'm very happy to share the model, I set layer f6/f4/f3 onlap stack relation type, so it looks one layer passes through other layers.
![image](https://github.com/gempy-project/gempy/assets/44236813/79ad3000-d9e9-4e6e-bb8a-c5a3ba58de9e)
![image](https://github.com/gempy-project/gempy/assets/44236813/0ed19a89-3fc2-4e87-a9e1-c4715b3e6a81)
It looks good, but somehow every layer has two colors, perhaps it's the boundary between up and down. The bottom layer which is basement looks weird, i wonder if it can be removed, but it doesn't matter if not, thanks for your quick response, i will come if i need help again.

**Reply by javoha** [reply]
Thanks for sharing!
Maybe first the one thing I can directly help with is the  "two color" thing - right now you are plotting the surfaces (opaque) and the lithology block, meaning lithology per grid cell (somewhat transparent). You can turn any of these off/on:
`gpv.plot_3d(data, show_lith=False, show_boundaries=True)`

You are right about orientations. There is just not one "right" way to get them. In gempy this is fully up to you. Best case you have actual data, but you can also manually pick any you want, or you can calculate them any way you like (similar to geomodeller).

Onlap relations can sometimes be a little tricky. If you like you can send me your code/notebook and I can take a closer look if I have ideas how to improve it. Definitely an interesting comparison.

**Reply by SpongeGirl** [reply]
I put my data and code in the zip file, column 'Z' in full.csv is obtained by column 'Z0' subtracts column 'To', column 'Z0' is copied by column 'Z' in VDH40Holes_Collars.csv which is in the Drilling.zip file, that means those points are above the f6 layer, i think you may know it by running the code, the orientation.csv file is obtained by geomodeller. Thanks for helping me, ask me if you have any questions.
[data&code.zip](https://github.com/gempy-project/gempy/files/15215756/data.code.zip)

**Reply by javoha** [reply]
Hi,
thanks for sharing the code. I took a quick look and changed your structural frame a little. Basically f1 is now in its own structural group (I had to add an upward facing orientation for that), f2 is also in its own structural group (still set to Erode as it erodes the underlying f1), while the f4 structural group is set to onlap. We dont see a real onlap though as the data does not really indicate it I guess. I wonder why the cross section you posted from geomodeller shows such a strong offset feature - again I dont really see it in the data here.

![image](https://github.com/gempy-project/gempy/assets/38279459/a28e3d5e-1366-4983-89ea-3052e88cd9a1)

And my code if you are interested (sorry, converted to a Jupyter and kept only minimum):

[data&code_adjsuted.zip](https://github.com/gempy-project/gempy/files/15221359/data.code_adjsuted.zip)

**Reply by SpongeGirl** [reply]
Nice of you, i think it's just an example of geomodeller, users can understand about onlap or erode data by this model, that's why it looks like a little bit weird, no need to think too much. Thanks very much for spending your time, i will close this discussion later.
By the way, i think lots of real data have no orientations, so maybe i should also do some research about Gemgis cause this tool supports more types of input data, like geological maps.
